File: platform/sdk/python/examples/t3_03_complete.py
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))
from xuanji_sdk_cloud import CloudClient

def main():
    client = CloudClient()
    bucket = "complete-bucket"
    key = "final/assembly.dat"
    r0 = client.createMultipartUpload(bucket, key)
    if not r0["ok"]:
        raise RuntimeError("create failed")
    uid = r0["upload_id"]
    parts = []
    total_len = 0
    for n in range(1, 4):
        s = f"PART-{n}-DATA-"
        total_len += len(s)
        r = client.uploadPart(bucket, key, uid, n, s)
        if not r["ok"]:
            raise RuntimeError(f"uploadPart {n} failed")
        parts.append({"part_number": n, "etag": r["etag"]})
    fin = client.completeMultipartUpload(bucket, key, uid, parts)
    if not fin["ok"]:
        raise RuntimeError("complete failed: " + str(fin))
    if not fin["etag"]:
        raise RuntimeError("final etag empty")
    obj = client.getObject(bucket, key)
    if not obj["found"]:
        raise RuntimeError("object not found after complete")
    data = obj["data"] if obj["data"] is not None else b""
    size = len(data)
    if size != total_len:
        raise RuntimeError(f"size mismatch expected={total_len} got={size}")
    print(f"XJ-OK: t3_03_complete etag={fin['etag']} size={size}")

if __name__ == "__main__":
    main()

File: platform/sdk/nodejs/examples/t3-03-complete.js
const { CloudClient } = require("../xuanji-sdk-cloud");

function main() {
  const client = new CloudClient();
  const bucket = "complete-bucket";
  const key = "final/assembly.dat";
  const r0 = client.createMultipartUpload(bucket, key);
  if (!r0.ok) throw new Error("create failed");
  const uid = r0.upload_id;
  const parts = [];
  let totalLen = 0;
  for (let n = 1; n <= 3; n++) {
    const s = "PART-" + n + "-DATA-";
    totalLen += s.length;
    const r = client.uploadPart(bucket, key, uid, n, s);
    if (!r.ok) throw new Error("uploadPart " + n + " failed");
    parts.push({ part_number: n, etag: r.etag });
  }
  const fin = client.completeMultipartUpload(bucket, key, uid, parts);
  if (!fin.ok) throw new Error("complete failed: " + fin.message);
  if (!fin.etag) throw new Error("final etag empty");
  const obj = client.getObject(bucket, key);
  if (!obj.found) throw new Error("object not found after complete");
  const size = (obj.data && obj.data.length) || 0;
  if (size !== totalLen) throw new Error("size mismatch expected=" + totalLen + " got=" + size);
  console.log("XJ-OK: t3-03-complete etag=" + fin.etag + " size=" + size);
}

if (require.main === module) main();
module.exports = main;
